### ISSUE TITLE
Don't send task names as strings

### ISSUE THREAD

**Issue by hds** [tokio-rs/console#462]
### What problem are you trying to solve?

Currently task names are sent as strings to tokio-console (the instrumentation client).

Instead they should be sent as metadata and the field value should be sent as a reference to the metadata element. This will reduce the data sent over the wire, especially in cases where a task with a fixed name is repeatedly spawned.

See comment for reference: https://github.com/tokio-rs/console/pull/452#discussion_r1316158519

### How should the problem be solved?

Send task names as metadata and reference the metadata in the spawn event.

**Comment by 0xPoe** [comment]
Hi @hds, I am interested in implementing this feature. I believe we need to update the Field to expose the ID, as currently, we can only retrieve its name from the public API. Or have I misunderstood this feature?

**Comment by hawkw** [comment]
@hi-rustin that's correct, this change will require an upstream change in `tracing` to expose the numeric index of a `tracing::field::Field`. I'm happy to merge a PR that makes such a change.

**Comment by hds** [comment]
@hi-rustin This shouldn't require any changes in `tracing`, it is only about how we send field names over the wire from `console-subscriber` to `tokio-console`.

The idea is that here (and in each of the other methods in this `impl`), instead of using the field name directly, we would look up the `Metadata` associated with `self.meta_id` and then search for the field name in the vec of field names. That's how we get the ID that we would use to create a `field::Name::NameIdx` variant.

https://github.com/tokio-rs/console/blob/075ab0c08f1c5f5b3648596466eb3ef7868e8b30/console-subscriber/src/visitors.rs#L309C28-L309C28

**Comment by 0xPoe** [comment]
> I'm happy to merge a PR that makes such a change.

Thanks for your reply! I've submitted https://github.com/tokio-rs/tracing/pull/2820.
